src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,731 @@
+            } else if (numPendingDeallocations != 0) {
+                // Get the number of queued deallocations that can be completed
+                uint256 queuedDeallocationIndicesLen =
+                    _queuedDeallocationIndices[operator][strategies[i]][operatorSetKey].length;
+
+                // index of pendingFreeMagnitude/deallocation to check for slashing
+                uint256 index = _queuedDeallocationIndices[operator][strategies[i]][operatorSetKey][queuedDeallocationIndicesLen - 1];
+                PendingFreeMagnitude storage pendingFreeMagnitude =
+                    _pendingFreeMagnitude[operator][strategies[i]][index];
+
+                // The most recent deallocation may not be completable and, therefore, slashable
+                if (block.timestamp < pendingFreeMagnitude.completableTimestamp) {
+                    // slash the pending deallocation proportionally and store the updated magnitudeDiff
+                    uint64 slashedAmount =
+                        uint64(uint256(bipsToSlash) * uint256(pendingFreeMagnitude.magnitudeDiff) / BIPS_FACTOR);
+                    pendingFreeMagnitude.magnitudeDiff -= slashedAmount;
+                    // keep track of total slashed magnitude
+                    slashedMagnitude += slashedAmount;
+                }
+            }
+
+            // 3. update totalMagnitude, get total magnitude and subtract slashedMagnitude
+            // this will be reflected in the conversion of delegatedShares to shares in the DM
+            Snapshots.History storage totalMagnitudes = _totalMagnitudeUpdate[operator][strategies[i]];
+            totalMagnitudes.push({key: uint32(block.timestamp), value: totalMagnitudes.latest() - slashedMagnitude});
+        }
+    }
+
+    /**
+     * @notice Called by an operator to cancel a salt that has been used to register with an AVS.
+     *
+     * @param salt A unique and single use value associated with the approver signature.
+     */
+    function cancelSalt(
+        bytes32 salt
+    ) external override {
+        // Mutate `operatorSaltIsSpent` to `true` to prevent future spending.
+        operatorSaltIsSpent[msg.sender][salt] = true;
+    }
+
+    /**
+     * @notice Helper for setting an operators allocation delay.
+     * @param operator The operator to set the delay on behalf of.
+     * @param delay The allocation delay in seconds.
+     */
+    function _setAllocationDelay(address operator, uint32 delay) internal {
+        require(delay != 0, InvalidDelay());
+
+        AllocationDelayInfo storage delayInfo = _allocationDelayInfo[operator];
+
+        bool pendingInEffect = delayInfo.pendingDelay != 0 && block.timestamp >= delayInfo.pendingDelayEffectTimestamp;
+
+        if (pendingInEffect) {
+            delayInfo.delay = delayInfo.pendingDelay;
+        }
+
+        delayInfo.pendingDelay = delay;
+        delayInfo.pendingDelayEffectTimestamp = uint32(block.timestamp + ALLOCATION_DELAY_CONFIGURATION_DELAY);
```

Comment:
This does 3 separate SSTORES.

Let's make this a `memory` variable and store it once at the end of this method.